ndb_api/db_helpers.go

```diff
@@ -165,6 +168,25 @@ func validateReqData(ctx context.Context, databaseInstanceType string, reqData m
 	return
 }
 
+// Converts a map to an action arguments list
+func convertMapToActionArguments(myMap map[string]string) []ActionArgument {
+	actionArgs := []ActionArgument{}
+	for name, value := range myMap {
+		actionArgs = append(actionArgs, ActionArgument{Name: name, Value: value})
+	}
+	return actionArgs
+}
+
+// Overwrites and appends actionArguments from database.additionalArguments to actionArguments
+func setConfiguredActionArguments(database DatabaseInterface, actionArguments map[string]string) {
+	allowedAdditionalArguments := util.GetAllowedAdditionalArgumentsForType(database.GetDBInstanceType())
+	for name, value := range database.GetDBInstanceAdditionalArguments() {
+		if isActionArgument, _ := allowedAdditionalArguments[name]; isActionArgument {
```

Comment:
Maybe add an explicit check to see if actionArguments is not nil? A nil pointer might cause a panic that might shut down the controller.

Reply:
Sounds good. The way it is programmed, actionArguments will never be nil (but adding the check is good practice). Will keep that in mind for the future.